src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/TilsynBeregningUtil.kt

```diff
@@ -58,4 +63,75 @@ object TilsynBeregningUtil {
             }
             .groupBy({ it.first }, { it.second }).mapValues { it.value.groupBy { it.type } }
     }
+
+    /**
+     * Splitter en periode opp i uker (kun hverdager inkludert)
+     * Tar inn en funksjon for å beregne hvor mange dager man ønsker å telle i uken
+     * enten alle eller begrense til antall aktivitetsdager
+     */
+    fun <P : Periode<LocalDate>> P.splitPerUke(antallDager: (fom: LocalDate, tom: LocalDate) -> Int): Map<Uke, PeriodeMedDager> {
+        val periode = mutableMapOf<Uke, PeriodeMedDager>()
+        var startOfWeek: LocalDate
+
+        if (this.fom.dayOfWeek == DayOfWeek.SATURDAY || this.fom.dayOfWeek == DayOfWeek.SUNDAY) {
+            startOfWeek = this.fom.with(TemporalAdjusters.next(DayOfWeek.MONDAY))
+        } else {
+            startOfWeek = this.fom.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
+        }
```

Comment:
Skjønte ikke behovet til denne først. 
Muligens man kan splitte ut den til en egen metode med kommentar? 
```kotlin
var startOfWeek: LocalDate = førsteDagIUka(this)
...

/**
     * Hvis man eks har lør-søn så blir første dagen i uka neste mandag som gjør at iterasjonen stopper
     */
    private fun Periode<LocalDate>.førsteDagIUka() =
        if (this.fom.dayOfWeek == DayOfWeek.SATURDAY || this.fom.dayOfWeek == DayOfWeek.SUNDAY) {
            this.fom.with(TemporalAdjusters.next(DayOfWeek.MONDAY))
        } else {
            this.fom.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
        }
```

Eller noe slik

Reply:
Ja kan splitte det ut for å gjøre så det blir litt tydligere! Fikk et par feil dersom måneden feks starta på en lørdag eller søndag 😢